daemon/api_systems.go

```diff
@@ -104,19 +105,45 @@ func getAllSystems(c *Command, r *http.Request, user *auth.UserState) Response {
 }
 
 // wrapped for unit tests
-var deviceManagerSystemAndGadgetInfo = func(dm *devicestate.DeviceManager, systemLabel string) (*devicestate.System, *gadget.Info, error) {
-	return dm.SystemAndGadgetInfo(systemLabel)
+var deviceManagerSystemAndGadgetAndEncryptionInfo = func(dm *devicestate.DeviceManager, systemLabel string) (*devicestate.System, *gadget.Info, *devicestate.EncryptionSupportInfo, error) {
+	return dm.SystemAndGadgetAndEncryptionInfo(systemLabel)
+}
+
+func storageEncryption(encInfo *devicestate.EncryptionSupportInfo) *client.StorageEncryption {
+	if encInfo.Disabled {
+		return &client.StorageEncryption{
+			Support: client.StorageEncryptionSupportDisabled,
+		}
+	}
+	storageEnc := &client.StorageEncryption{
+		StorageSafety: string(encInfo.StorageSafety),
+		Type:          string(encInfo.Type),
+	}
+	required := (encInfo.StorageSafety == asserts.StorageSafetyEncrypted)
+	switch {
+	case encInfo.Available:
+		storageEnc.Support = client.StorageEncryptionSupportAvailable
+	case !encInfo.Available && required:
+		storageEnc.Support = client.StorageEncryptionSupportDefective
+		storageEnc.UnavailableReason = encInfo.UnavailableErr.Error()
+	case !encInfo.Available && !required:
+		storageEnc.Support = client.StorageEncryptionSupportUnavailable
+		storageEnc.UnavailableReason = encInfo.UnavailableWarning
```

Comment:
this one is tested though.

We need more cases or a dedicated test for this function.